modin/pandas/series.py

```diff
@@ -1196,6 +1196,12 @@ def interpolate(
             **kwargs,
         )
 
+    def isin(self, values):  # noqa: PR01, RT01, D200
+        """
+        Whether elements in `Series` are contained in `values`.
+        """
+        return super(Series, self).isin(values, shape_hint="column")
```

Comment:
We generally can't predict the shape hint at this layer. The whole point of the `.columnarize()` function is to support backends that may store columns row-wise thus representing a single-column Series as a single-row frame. Modin has to be backend-agnostic at the front-end meaning that we can't dictate this `shape_hint` parameter from here, it should be handled at the query compiler level I guess.

Reply:
We can predict the shape for Series.isin because it always returns a Series of booleans. We can pass hints from the higher layers in to lower layers.

Reply:
Let's take this change down a level (into `PandasQueryCompiler`), as we did for the rest of the functions.

Reply:
In https://github.com/modin-project/modin/pull/5658 we did such a change in `PandasQC` because we did know that the underlying object is a Series. But for `isin` at the QC layer we don't have the knowledge if the underlying object is a Series or a DataFrame.

Reply:
Should we store `_pandas_class` at the query compiler level for that?

Reply:
Not sure what you mean and what for? Elaborate please.

Reply:
Based on the type stored in `_pandas_class` variable, it could be possible to pass `shape_hint` parameter or not to the constructor of the query compiler.

Reply:
`_pandas_class` is a variable of the high-level objects, namely, Series, DataFrame, BasePandasDataset. We could pass that variable to the QC instead of the direct hint `shape_hint="column"` and then determine if the underlying object is a DataFrame or Series. However, I think it is an extra step.

Reply:
On second thought, it really seems that the `shape_hint` parameter makes sense for now

Reply:
ok for me